### DISCUSSION TITLE
Kitty remains active after exiting an X11 session

### DISCUSSION THREAD

**Discussion by b02860de585071a2** [kovidgoyal/kitty#5728] [Q&A] [answered]
When closing an X session (ie with `pkill -15 Xorg`), the kitty process continues to run in the tty session, and keeps all Nvidia modules active/loaded despite not being actively in use. This includes with shell integration and remote control both disabled, and no other related features active.

Since the process shows as "kitty : bash|/bin/bash --posix" in top, I assume this has something to do with its bash hooks, or maybe it's keeping modules loaded for hardware acceleration? I looked over the shell integration article, but didn't see anything particularly obvious as to why it would still be running.

The reason I ask is because it creates unique issues with VFIO, or when rebuilding kernel modules with `mkinitcpio`. You cannot `modprobe -r` to unhook them until kitty has been manually killed with a `SIGTERM` or `SIGKILL`.

It's not hard to add "`pkill -15 kitty`" to my exit processes, but I'm curious as to why this is the case. I do a fair bit of work in the tty, and may bounce between that and a VM, so it's a a minor annoyance.

Here is the output of [lsof | grep -i nvidia](https://gist.github.com/FAT9L/e578800a0db339a50ebcb61441ab5171) from a tty after killing X.

**Comment by kovidgoyal** [comment, marked as answer]
kitty does not listen for the X server to quit. Indeed, as far as I know
there is no mechanism where the X server informs clients it has quit. 

It will only quit if it needs to make some request to the X server and
in so doing discovers that the connection to the server has been broken.

And normally, kitty will sit waiting for input not making any requests
to the X server consuming 0 CPU.

So you need to kill kitty explicitly.

**Reply by b02860de585071a2** [reply]
Thank you for the explanation. 

I guess it's a misunderstanding on my part - every other terminal emulator I've used is destroyed when X closes a session + takes down any graphical processes that were spawned.
